node/controllers/index.js

```diff
@@ -19,30 +18,66 @@ function postHandler(req, res) {
       attachments: util.createOutputAttachments(req.body.text, "#3AA3E3")
     })
 
-    lib.runCode(lang, sanitized, req.body.trigger_id)
-      .then(res => {
+    if (lang === 'python' || lang === 'cpp' || lang === 'bash') {
+      cachedCode(req, lang, sanitized)
+    } else {
+      runCode(req, lang, sanitized)
+    }
+
+  }
+}
+
+async function cachedCode(req, lang, sanitized) {
+  try {
+    const [ error, reply ] = await redisGet(sanitized)
```

Comment:
Because you're handling an error in the catch the same way it should be fine if you reject from redisGet